docs/source/delayed.rst

```diff
@@ -209,3 +209,14 @@ on the side-effect of another task. In these cases you can use
 ``sum_data`` will operate on DATA only after both the expected items have
 been appended to it. ``bind`` can also be used along with direct dependencies
 passed through the function arguments.
+
+
+Execution
+---------
+
+By default, Dask Delayed uses the threaded scheduler in order to avoid data
+transfer costs.  
+You should consider using multiprocessing or `dask.distributed`_ scheduler 
+on a local machine or on a cluster if your code does not release the GIL well.
```

Comment:
I wonder if we can add a link to some docs that explain what this mean, I only found this but not sure if it's enough https://docs.dask.org/en/stable/graphs.html#avoid-holding-the-gil

Reply:
Here again, I only copy pasted something. I just know that multi-threading with pure Python code or libraries that hold to the GIL won't work. 

There are some mention of the GIL here too: https://docs.dask.org/en/stable/scheduling.html#local-processes.